src/stats.jl

```diff
@@ -1,18 +1,25 @@
 #################### Posterior Statistics ####################
 
 """
-    autocor(chains[; lags = [1, 5, 10, 50], demean = true, append_chains = true, kwargs...])
+    autocor(
+        chains;
+        append_chains = true,
+        demean = true,
+        [lags,]
+        kwargs...,
+    )
 
 Compute the autocorrelation of each parameter for the chain.
 
-Setting `append_chains=false` will return a vector of dataframes containing the
-autocorrelations for each chain.
+The default `lags` are `[1, 5, 10, 50]`, upper-bounded by `n - 1` where `n` is the number of samples used in the estimation.
+
+Setting `append_chains=false` will return a vector of dataframes containing the autocorrelations for each chain.
 """
 function autocor(
     chains::Chains;
-    lags::AbstractVector{<:Integer} = [1, 5, 10, 50],
+    append_chains = true,
```

Comment:
This was inconsistent with all other functions. Therefore also the tests didn't seem reasonable since they only tested `append_chains=false`.